testutils/utils.go

```diff
@@ -250,17 +242,30 @@ func CreateContainerNamespaceWithCid(container_id string) (containerNs ns.NetNS,
 }
 
 func CreateContainer(netconf, podName, podNamespace string, ip string) (container_id string, session *gexec.Session, contVeth netlink.Link, contAddr []netlink.Addr, contRoutes []netlink.Route, targetNs ns.NetNS, err error) {
-	return CreateContainerWithId(netconf, podName, podNamespace, ip, "")
```

Comment:
Not for this PR, but for future clean up - these functions are VERY similar, and both take a lot of arguments and return even more.

It would be great to use structs for argument and return values.

e.g. 

```go
func CreateContainer(args CreateArgs) (ContainerParams, error) {
   . . .
}

type CreateArgs struct {
    NetConf string
    PodName string
    PodNamespace string
    IP string
    ContainerID string // Optional
}

type ContainerParams struct {
    ID string
    Session *gexec.Session
    Veth netlink.Link
    Addrs []Addrs
    Routes []Route
    Namespace ns.NetnS
}
```

Not a trivial amount of work, but I think could clean up the code nicely if you're ever bored.

Reply:
Agreed, I'm working towards that.

Reply:
I actually think `CreateContainer` should be removed entirely as well, and test flow should look like this:

1. Create & configure container ns, check for errors
2. Run calico plugin against it, check for errors

By combining those two things in a single function like CreateContainer does, the error from 1 and 2 are less distinguishable. It will introduce some redundancy in the tests but I think that can be resolved by moving it into beforeEach.

Reply:
Sounds fine to me